### PR TITLE
Remove knowledge of child StateT type from compose.

### LINKED ISSUE #213: WorkflowContext.compose shouldn't have to know child workflow's state type.
This was only required back when we internally keyed off of the tuple of type parameters. Since we use the actual `KClass` of `Workflow` implementations now, we don't need to care anymore.

### PR DESCRIPTION
**Pull request by zach-klippenstein**
Closes #213.

### CODE REVIEW

File: kotlin/workflow-host/src/main/java/com/squareup/workflow/internal/WorkflowNode.kt [hunk 1]
```diff
@@ -38,7 +38,7 @@ import kotlin.coroutines.experimental.CoroutineContext
  * [Workflow.initialState].
  */
 internal class WorkflowNode<InputT : Any, StateT : Any, OutputT : Any, RenderingT : Any>(
```
Comment (rjrjr) [inline kotlin/workflow-host/src/main/java/com/squareup/workflow/internal/WorkflowNode.kt]: If `WorkflowContext.compose` no longer knows about `StateT`, how is `WorkflowNode` learning about it? Where does it come from?

Comment (zach-klippenstein) [reply kotlin/workflow-host/src/main/java/com/squareup/workflow/internal/WorkflowNode.kt]: It basically doesn't, it's effectively `Any`. But it doesn't need to in practice. The generic param is basically just there so the `Workflow` methods all agree.

File: kotlin/workflow-host/src/main/java/com/squareup/workflow/internal/WorkflowNode.kt [hunk 2]
```diff
@@ -91,22 +91,12 @@ internal class WorkflowNode<InputT : Any, StateT : Any, OutputT : Any, Rendering
...
-
-    behavior = context.buildBehavior()
-    // Start new children/subscriptions, and drop old ones.
-    subtreeManager.track(behavior.childCases)
-    subscriptionTracker.track(behavior.subscriptionCases)
-
-    return rendering
-  }
+  ): RenderingT =
+    composeWithStateType(workflow as Workflow<InputT, StateT, OutputT, RenderingT>, input)
```
Comment (zach-klippenstein) [inline kotlin/workflow-host/src/main/java/com/squareup/workflow/internal/WorkflowNode.kt]: @rjrjr The cast you're looking for is here.

File: kotlin/workflow-host/src/main/java/com/squareup/workflow/internal/WorkflowNode.kt (hunk 1 above)
Comment (rjrjr) [reply kotlin/workflow-host/src/main/java/com/squareup/workflow/internal/WorkflowNode.kt]: Let me see if I understand: the `StateT` of the root `WorkflowNode` comes from `WorkflowHost.Factory#run$workflow_host`. But of every other `WorkflowNode` in the tree has `StateT` of `Any`? Given that, can you push this PR farther and get rid of `StateT` from `WorkflowNode` and `RealWorkflowContext`?

**Comment by zach-klippenstein** [comment]
Declining in favor of #219.

**Pull request closed without merging**